inginious/frontend/pages/register.py

```diff
@@ -101,7 +106,11 @@ def register_user(self, data):
                                             "password": passwd_hash,
                                             "activate": activate_hash,
                                             "bindings": {},
-                                            "language": self.user_manager._session.get("language", "en")})
+                                            "language": self.user_manager._session.get("language", "en"),
+                                            })
+                if self.is_tos_defined():
+                    self.database.users.find_one_and_update({"username": data["username"]},
```

Comment:
This could probably be simplified and put in one single insert request.

    ...
    "tos_accepted": self.is_tos_defined() and "term_policy_check" in data
    ...